static-assets/components/cstudio-dialogs/uploadCMIS-dialog.js

```diff
@@ -0,0 +1,290 @@
+        this.path = path;
+        this.profile = repositoryId;
+        this.asPopup = true;
+        this.serviceUri = serviceUri;
+        this.callback = callback;
+        this.dialog.show();
+        document.getElementById("cstudio-wcm-popup-div_h").style.display = "none";
+
+        if(window.frameElement){
+            var id = window.frameElement.getAttribute("id").split("-editor-")[1];
+            var getFormSizeVal = typeof getFormSize === 'function' ? getFormSize : parent.getFormSize;
+            var setFormSizeVal = typeof setFormSize === 'function' ? setFormSize : parent.setFormSize;
+            var formSize = getFormSizeVal(id);
+            if(formSize < 320){
+                setFormSizeVal(320, id);
+                $($(".studio-ice-container-"+id,parent.document)[0]).attr('data-decrease', true);
+            }
+        }
+    },
+
+    /**
+     * hide dialog
+     */
+    closeDialog:function() {
+        this.dialog.destroy();
+    },
+
+    /**
+     * create dialog
+     */
+    createDialog: function(path, site, repositoryId, serviceUri) {
+        var me = this;
+        YDom.removeClass("cstudio-wcm-popup-div", "yui-pe-content");
+
+        var newdiv = YDom.get("cstudio-wcm-popup-div");
+        if (newdiv == undefined) {
+            newdiv = document.createElement("div");
+            document.body.appendChild(newdiv);
+        }
+
+        var divIdName = "cstudio-wcm-popup-div";
+        newdiv.setAttribute("id",divIdName);
+        newdiv.className= "yui-pe-content";
+        newdiv.innerHTML = '<div class="contentTypePopupInner" id="upload-popup-inner">' +
+            '<div class="contentTypePopupContent" id="contentTypePopupContent"> ' +
+            '<div class="contentTypePopupHeader">Upload</div> ' +
+            '<div><form id="asset_upload_form">' +
+            '<div class="contentTypeOuter">'+
+            '<div class="formDesc">Please select a file to upload</div> ' +
+            '<div><table><tr><td><input type="hidden" name="site_id" value="' + site + '"/></td>' +
+            '<td><input type="hidden" name="cmis_path" value="' + path + '"/></td></tr>' +
+            '<td><input type="hidden" name="cmis_repo_id" value="' + repositoryId + '"/></td></tr>' +
+            '<tr><td>File:</td><td><input type="file" name="file" id="uploadFileNameId"/></td></tr>' +
+            '</table></div>' +
+            '</div>' +
+            '<div class="contentTypePopupBtn"> ' +
+            '<input type="button" class="btn btn-primary cstudio-xform-button ok" id="uploadButton" value="Upload" disabled />' +
+            '<input type="button" class="btn btn-default cstudio-xform-button" id="uploadCancelButton" value="Cancel"  /></div>' +
+            '</form></div>' +
+            '<div><div  style="visibility:hidden; margin-bottom:1.5em;" id="indicator">Uploading...</div>' +
+            '</div> ' +
+            '</div>';
+
+        document.getElementById("upload-popup-inner").style.width = "350px";
```

Comment:
Since these are hard coded, why not set them directly on the HTML. But better yet, if they're not calculated or dynamic set these styles on a stylesheet to the class.

Reply:
Done